File: 2395. Find Subarrays With Equal Sum.php
class Solution {
    function findSubarrays($nums) {
    for ($i = 0; $i < count($nums) - 2; $i++) {
        for ($j = count($nums) - 2; $j >= $i + 1; $j--) {
            if ($nums[$i] + $nums[$i + 1] == $nums[$j] + $nums[$j + 1]) {
                return true;
            }
        }
    }
    return false;
    }
}

File: 2395. Find Subarrays With Equal Sum.rs
impl Solution {
    pub fn find_subarrays(nums: Vec<i32>) -> bool {
    for i in 0..nums.len()-2 {
        for j in (i+1..nums.len()-1).rev() {
            if nums[i] + nums[i+1] == nums[j] + nums[j+1] {
                return true;
            }
        }
    }
    false 
    }
}
